tests/test_io.py

```diff
@@ -0,0 +1,229 @@
+    )
+    assert reference.equals(dataframe)
+
+
+def test_get_grib_groups():
+    reference = pd.DataFrame(
+        {
+            "feature_name": [
+                "u_900.0",
+                "u_50.0",
+                "v_900.0",
+                "v_50.0",
+                "u_900.0",
+                "u_50.0",
+                "v_900.0",
+                "v_50.0",
+            ],
+            "level": [900.0, 50.0, 900.0, 50.0, 900.0, 50.0, 900.0, 50.0],
+            "name": ["u", "u", "v", "v", "u", "u", "v", "v"],
+            "typeOfLevel": [
+                "heightAboveGround",
+                "heightAboveGround",
+                "heightAboveGround",
+                "heightAboveGround",
+                "isobaricInhPa",
+                "isobaricInhPa",
+                "isobaricInhPa",
+                "isobaricInhPa",
+            ],
+        }
+    )
+
+    groups = wr.get_grib_groups(reference)
+
+    assert set(groups.keys()) == {
+        "u_heightAboveGround",
+        "v_heightAboveGround",
+        "u_isobaricInhPa",
+        "v_isobaricInhPa",
+    }
+
+    assert groups["u_heightAboveGround"]["level"] == [900.0, 50.0]
+    assert groups["u_heightAboveGround"]["cfVarName"] == "u"
+    assert groups["u_heightAboveGround"]["typeOfLevel"] == "heightAboveGround"
+
+    assert groups["v_heightAboveGround"]["level"] == [900.0, 50.0]
+    assert groups["v_heightAboveGround"]["cfVarName"] == "v"
+    assert groups["v_heightAboveGround"]["typeOfLevel"] == "heightAboveGround"
+
+    assert groups["u_isobaricInhPa"]["level"] == [900.0, 50.0]
+    assert groups["u_isobaricInhPa"]["cfVarName"] == "u"
+    assert groups["u_isobaricInhPa"]["typeOfLevel"] == "isobaricInhPa"
+
+    assert groups["v_isobaricInhPa"]["level"] == [900.0, 50.0]
+    assert groups["v_isobaricInhPa"]["cfVarName"] == "v"
+    assert groups["v_isobaricInhPa"]["typeOfLevel"] == "isobaricInhPa"
+
+
+@dataclass
+class FakeSample:
+    date: str
+    fancy_ident: str
+
+
+def test_get_output_filename():
+
+    saving_settings = wr.GribSavingSettings(
+        template_grib="fake_grib.grib",
+        directory="fakedirectory",
+        sample_identifiers=("date", "leadtime"),
+        output_fmt="grid.forecast_ai_date_{}_ech_{}.json",
+    )
+    sample = FakeSample(date="1911-10-30", fancy_ident="first solvay congress")
+    leadtime = 1.0
+
+    filename = wr.get_output_filename(saving_settings, sample, leadtime)
+    assert filename == f"grid.forecast_ai_date_1911-10-30_ech_{leadtime}.json"
+
+    saving_settings = wr.GribSavingSettings(
+        template_grib="fake_grib.grib",
+        directory="fakedirectory",
+        output_kwargs=("congress",),
+        sample_identifiers=("date", "leadtime"),
+        output_fmt="grid.forecast_{}_date_{}_ech_{}.json",
+    )
+
+    filename = wr.get_output_filename(saving_settings, sample, leadtime)
+
+
+if __name__ == "__main__":
+    test_nan_mask()
+    test_get_grib_param_dataframe()
+    test_get_grib_groups()
```

Comment:
Running tests should preferably be done using pytest cli:

```bash
 runai exec python -m pytest tests/test_io.py
 ```

Reply:
Yup, that what to test the tests themselves :)

Reply:
but now it's gone :)